app/gist/edit/route.js

```diff
@@ -7,6 +8,16 @@ export default GistRoute.extend({
     return this.store.find('gist', params.id);
   },
 
+  setupController() {
+    this._super.apply(this, arguments);
+
+    let gistController = this.controllerFor('gist');
+    gistController.rebuildApp();
+    Ember.run.schedule('afterRender', function() {
+      gistController.set('unsaved', false);
+    });
+  },
+
```

Comment:
So the observer was synchronous, while the action waits for it's turn in the 'actions' queue. We need to set 'unsaved' to false after the action has taken place.